tests/unit/s2n_send_key_update_test.c

```diff
@@ -165,16 +169,16 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_negotiate_test_server_and_client(server_conn, client_conn));
 
         /* Force Client to send a TLS 1.3 KeyUpdate Message over TLS 1.2 connection */
```

Comment:
Why are we checking this ? Shouldn't a KeyUpdate message never be sent over <TLS1.3 or I am misunderstanding something.

Reply:
This is testing the receive side, not the send side. A KeyUpdate should never be sent with TLS1.2, but we can't control what our peer does. If our peer sends us a KeyUpdate with TLS1.2, we need to gracefully detect it, return an error, and close the connection.